core/src/components/checkbox/checkbox.ionic.scss

```diff
@@ -319,17 +326,21 @@ input {
 
 :host(.checkbox-disabled) .native-wrapper:after {
   @include globals.disabled-state();
+  @include globals.position(unset, unset, unset, unset);
+
+  width: inherit;
+  height: inherit;
```

Comment:
These styles are required in order to properly cover the border of the checkbox element. When `top` and `left` are set, it is calculated from the [beginning of the content](https://www.w3.org/TR/CSS21/visudet.html#containing-block-details), inside of the border. The alternative solution I found was calculating the position based on the `--border-width` CSS variable:

```scss
@include globals.position(
    calc(-1 * var(--border-width)), 
    calc(-1 * var(--border-width)), 
    calc(-1 * var(--border-width)), 
    calc(-1 * var(--border-width))
);
```

Inheriting the width and height is a cleaner solution, in my opinion. I could update the `disabled-state` mixin to remove `position` and inherit `width` and `height` as well, but I wasn't sure if this would have any negative consequences on other components.

Reply:
I think the inherit solution is much more clean yes 💪.

About the mixin, I'm not so sure, I think we will need the position on some cases.

For now is only being used on Checkbox, as other components were made before the mixin, but I did a quick test on the Chip and it works correctly by adding the mixin as it is now, but when I replaced the position with the width/height inherit, it didn't work, as the size was 0 🫠

Reply:
Hmm okay. We can just add this only for checkbox. I can add some comments as to why it is different than the mixin. Then we can look into if we need to change it later.

Reply:
Added a comment explaining why we override the mixin: https://github.com/ionic-team/ionic-framework/pull/29623/commits/3bf7ea3f22dd6fbe98f942c21773d6ef6acc1d54